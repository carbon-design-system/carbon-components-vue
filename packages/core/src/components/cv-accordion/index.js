import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
