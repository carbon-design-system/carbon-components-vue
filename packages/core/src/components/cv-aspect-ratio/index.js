import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
