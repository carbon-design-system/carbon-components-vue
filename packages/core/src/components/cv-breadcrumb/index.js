import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
