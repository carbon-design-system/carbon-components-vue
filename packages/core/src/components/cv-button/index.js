import CvButton from './cv-button';
import CvIconButton from './cv-icon-button';
import CvButtonSkeleton from './cv-button-skeleton';
import CvButtonSet from './cv-button-set';

export { CvButton, CvIconButton, CvButtonSkeleton, CvButtonSet };
