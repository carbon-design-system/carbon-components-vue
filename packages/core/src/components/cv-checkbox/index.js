import CvCheckbox from './cv-checkbox';
import CvCheckboxSkeleton from './cv-checkbox-skeleton';
export { CvCheckbox, CvCheckboxSkeleton };
