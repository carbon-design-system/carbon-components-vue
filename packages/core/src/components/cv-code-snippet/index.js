import CvCodeSnippet from './cv-code-snippet';
import CvCodeSnippetSkeleton from './cv-code-snippet-skeleton';
export { CvCodeSnippet, CvCodeSnippetSkeleton };
