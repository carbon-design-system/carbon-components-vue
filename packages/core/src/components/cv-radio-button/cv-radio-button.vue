<template>
  <div
    :class="[
      `cv-radio-button ${carbonPrefix}--radio-button-wrapper`,
      { [`${carbonPrefix}--radio-button-wrapper--label-left`]: labelLeft },
    ]"
  >
    <input
      v-bind="$attrs"
      v-on="inputListeners"
      :id="uid"
      :checked="isChecked"
      :class="`${carbonPrefix}--radio-button`"
      ref="input"
      type="radio"
      :value="value"
    />
    <!-- symbol causes problem in codepen? -->
    <label :for="uid" :class="`${carbonPrefix}--radio-button__label`">
      <span :class="`${carbonPrefix}--radio-button__appearance`"></span>
      {{ label }}
    </label>
  </div>
</template>

<script>
import { uidMixin, radioMixin, carbonPrefixMixin, methodsMixin } from '../../mixins';

export default {
  name: 'CvRadioButton',
  inheritAttrs: false,
  mixins: [uidMixin, radioMixin, carbonPrefixMixin, methodsMixin({ input: ['blur', 'focus'] })],
  props: {
    labelLeft: Boolean,
  },
};
</script>
