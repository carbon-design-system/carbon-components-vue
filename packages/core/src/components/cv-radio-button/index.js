import CvRadioGroup from './cv-radio-group';
import CvRadioButton from './cv-radio-button';

export { CvRadioGroup, CvRadioButton };
