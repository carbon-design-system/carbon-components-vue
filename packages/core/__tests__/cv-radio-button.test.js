// import { shallowMount as shallow } from '@vue/test-utils';
import { testComponent, awaitNextTick } from './_helpers';
const { shallowMount: shallow } = awaitNextTick;
import { CvRadioButton, CvRadioGroup } from '@/components/cv-radio-button/';

describe('CvRadioButton', () => {
  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(CvRadioButton, ['labelLeft', 'checked'], Boolean);
  testComponent.propsAreType(CvRadioButton, ['modelValue', 'value', 'label'], String);
  testComponent.propsAreRequired(CvRadioButton, ['value']);

  // ***************
  // SNAPSHOT CHECKS
  // ***************
  it('matches render with right label', async () => {
    const wrapper = await shallow(CvRadioButton, {
      propsData: {
        label: 'right label',
        labelLeft: false,
        id: '1',
        value: '',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches render with left label', async () => {
    const wrapper = await shallow(CvRadioButton, {
      propsData: {
        label: 'left label',
        labelLeft: true,
        id: '1',
        value: '',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches render when checked', async () => {
    const wrapper = await shallow(CvRadioButton, {
      propsData: {
        label: 'test label',
        labelLeft: false,
        id: '1',
        value: '',
        checked: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches render when modelValue is equal to value', async () => {
    const wrapper = await shallow(CvRadioButton, {
      propsData: {
        label: 'test label',
        labelLeft: false,
        id: '1',
        value: 'test value',
        modelValue: 'test value',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches render when modelValue is not equal to value', async () => {
    const wrapper = await shallow(CvRadioButton, {
      propsData: {
        label: 'test label',
        labelLeft: false,
        id: '1',
        value: 'test value',
        modelValue: 'not test value',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL CHECKS
  // ***************
  it('is not checked when checked and modelValue props are not set', async () => {
    const wrapper = await shallow(CvRadioButton, {
      propsData: {
        labelLeft: false,
        id: '1',
        value: 'test value',
      },
    });
    expect(wrapper.vm.isChecked).toEqual(false);
  });

  it('is checked when checked prop is set to true and modelValue prop is not set', async () => {
    const wrapper = await shallow(CvRadioButton, {
      propsData: {
        labelLeft: false,
        id: '1',
        value: 'test value',
        checked: true,
      },
    });
    expect(wrapper.vm.isChecked).toEqual(true);
  });

  it('modelValue prop has higher priority than checked prop', async () => {
    const wrapper = await shallow(CvRadioButton, {
      propsData: {
        labelLeft: false,
        id: '1',
        value: 'test value',
        checked: true,
        modelValue: 'not test value',
      },
    });
    expect(wrapper.vm.isChecked).toEqual(false);
  });
});

describe('CvRadioGroup', () => {
  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(CvRadioGroup, ['vertical'], Boolean);

  // ***************
  // SNAPSHOT CHECKS
  // ***************
  it('matches render when vertical', async () => {
    const wrapper = await shallow(CvRadioGroup, {
      propsData: { vertical: true },
      slots: {
        default: '<div class="cv-radio-button-stub">StubbyMcStubFace</div>',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches render when horizontal', async () => {
    const wrapper = await shallow(CvRadioGroup, {
      propsData: { vertical: false },
      slots: {
        default: '<div class="cv-radio-button-stub">StubbyMcStubFace</div>',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL CHECKS
  // ***************
});
