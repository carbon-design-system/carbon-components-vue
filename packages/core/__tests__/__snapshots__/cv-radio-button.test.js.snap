// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvRadioButton matches render when checked 1`] = `
<div class="cv-radio-button bx--radio-button-wrapper"><input id="1" type="radio" class="bx--radio-button" value=""> <label for="1" class="bx--radio-button__label"><span class="bx--radio-button__appearance"></span> <span class="">
      test label
    </span></label></div>
`;

exports[`CvRadioButton matches render when modelValue is equal to value 1`] = `
<div class="cv-radio-button bx--radio-button-wrapper"><input id="1" type="radio" class="bx--radio-button" value="test value"> <label for="1" class="bx--radio-button__label"><span class="bx--radio-button__appearance"></span> <span class="">
      test label
    </span></label></div>
`;

exports[`CvRadioButton matches render when modelValue is not equal to value 1`] = `
<div class="cv-radio-button bx--radio-button-wrapper"><input id="1" type="radio" class="bx--radio-button" value="test value"> <label for="1" class="bx--radio-button__label"><span class="bx--radio-button__appearance"></span> <span class="">
      test label
    </span></label></div>
`;

exports[`CvRadioButton matches render with left label 1`] = `
<div class="cv-radio-button bx--radio-button-wrapper bx--radio-button-wrapper--label-left"><input id="1" type="radio" class="bx--radio-button" value=""> <label for="1" class="bx--radio-button__label"><span class="bx--radio-button__appearance"></span> <span class="">
      left label
    </span></label></div>
`;

exports[`CvRadioButton matches render with right label 1`] = `
<div class="cv-radio-button bx--radio-button-wrapper"><input id="1" type="radio" class="bx--radio-button" value=""> <label for="1" class="bx--radio-button__label"><span class="bx--radio-button__appearance"></span> <span class="">
      right label
    </span></label></div>
`;

exports[`CvRadioGroup matches render when horizontal 1`] = `
<div class="cv-radio-group bx--form-item">
  <div class="bx--radio-button-group">
    <div class="cv-radio-button-stub">StubbyMcStubFace</div>
  </div>
</div>
`;

exports[`CvRadioGroup matches render when vertical 1`] = `
<div class="cv-radio-group bx--form-item">
  <div class="bx--radio-button-group bx--radio-button-group--vertical">
    <div class="cv-radio-button-stub">StubbyMcStubFace</div>
  </div>
</div>
`;
